analytics/components/Plausible.tsx

```diff
@@ -0,0 +1,104 @@
+
+const exclusionScript =
+  '!function(){"use strict";var o=window.location,l=window.document,s=l.currentScript,p=s.getAttribute("data-api")||new URL(s.src).origin+"/api/event";function c(t,e){t&&console.warn("Ignoring Event: "+t),e&&e.callback&&e.callback()}function t(t,e){if(/^localhost$|^127(\\.[0-9]+){0,2}\\.[0-9]+$|^\\[::1?\\]$/.test(o.hostname)||"file:"===o.protocol)return c("localhost",e);if(window._phantom||window.__nightmare||window.navigator.webdriver||window.Cypress)return c(null,e);try{if("true"===window.localStorage.plausible_ignore)return c("localStorage flag",e)}catch(t){}var i=s&&s.getAttribute("data-include"),n=s&&s.getAttribute("data-exclude");if("pageview"===t){i=!i||i.split(",").some(a),n=n&&n.split(",").some(a);if(!i||n)return c("exclusion rule",e)}function a(t){return o.pathname.match(new RegExp("^"+t.trim().replace(/\\*\\*/g,".*").replace(/([^\\.])\\*/g,"$1[^\\\\s/]*")+"/?$"))}var i={},r=(i.n=t,i.u=o.href,i.d=((w,d)=>{const h=w.location.hostname;return h.replace(/^www\./,"")})(window,document),i.r=l.referrer||null,e&&e.meta&&(i.m=JSON.stringify(e.meta)),e&&e.props&&(i.p=e.props),new XMLHttpRequest);r.open("POST",p,!0),r.setRequestHeader("Content-Type","text/plain"),r.send(JSON.stringify(i)),r.onreadystatechange=function(){4===r.readyState&&e&&e.callback&&e.callback()}}var e=window.plausible&&window.plausible.q||[];window.plausible=t;for(var i,n=0;n<e.length;n++)t.apply(this,e[n]);function a(){i!==o.pathname&&(i=o.pathname,t("pageview"))}var r,u=window.history;u.pushState&&(r=u.pushState,u.pushState=function(){r.apply(this,arguments),a()},window.addEventListener("popstate",a)),"prerender"===l.visibilityState?l.addEventListener("visibilitychange",function(){i||"visible"!==l.visibilityState||a()}):a()}();';
+
+const plausibleScript = exclusionScript;
+
+const sendEvent = (
```

Comment:
Add a comment, that this function should be self contained, because it is `Stringified`